.ci/checkTranslator.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Load colors and such
+SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source "$SCRIPT_DIR/helper.sh"
+
+# Global variable holding the currently checked file
+TRANSLATOR=
+TRANSLATOR_BASENAME=
+
+# These checks will result in an error
+declare -a ERROR_CHECKS=(
+    "notOneTranslatorID"
+    "nonUniqueTranslatorID"
+    "reusingDeletedID"
+    "executableFile"
+    "invalidJSON"
+    "withCarriageReturn"
+    "deprecatedForEach"
+)
+
+# These checks will emit warnings but still pass
+declare -a WARN_CHECKS=(
+    "badLicense"
+    "problematicJS"
+)
+
+#-----------------------------------------------------------------------
+# Checks
+#-----------------------------------------------------------------------
+
+# Every translator must have exactly one translator ID
+notOneTranslatorID () {
+    local nIds
+    nIds=$(grep -c '"translatorID"' "$TRANSLATOR")
+    case "$nIds" in
+        1) return ;;
+        0) err "Missing translatorID" ;;
+        *) err "Too many translatorID"
+    esac
+    return 1
+}
+
+# Every translatorID must be unique
+nonUniqueTranslatorID () {
+    local dir id duplicateIds
+    dir=$(dirname "$TRANSLATOR")
+    id=$(grepTranslatorId "$TRANSLATOR")
+    if [[ -n "$id" ]];then
+      duplicateIds=$(grepTranslatorId "$dir"/*.js | sort | uniq -d)
```

Comment:
Actually, this will be performed for each translator, which seems to be then too much, because we already know the specific `id` to look for. I would suggest something like
```bash
duplicateIds=$(grep "$id" | uniq -d)
```
which can then be tested to being empty.